### ISSUE TITLE
Can not do CLI mount 

### ISSUE THREAD

**Issue by thyeun** [eclipse-che/che#2123]
Not possible to done the **Che mount**, it keep on given an error **path does not exist**, as you can see from the picture below. (i have the correct ws-ssh-port, and "mount" folder)

OS : windows 10 build 10240
Che Version : 4.6.2

![s9](https://cloud.githubusercontent.com/assets/1939446/17656712/bac621aa-62ef-11e6-8d74-8db3729c1631.JPG)

**Comment by ghost** [comment]
@thyeun what if you provide this path as `c/users/thomas/mount`? Does it make any difference.

**Comment by TylerJewell** [comment]
We convert directory names from the c: format to the one you describe, Eugene.  The issue is virtualbox does not allow the directory provided by @thyuen to be mounted.  It is a virtualbox problem.

**Comment by thyeun** [comment]
@TylerJewell and @eivantsov, here goes all the try as show from the below pictures, but non of it success, but one of them have a different error (**can not enable tty mode**)

![s11](https://cloud.githubusercontent.com/assets/1939446/17663621/6c61d226-6320-11e6-92af-f157acebd41a.JPG)

**Comment by ghost** [comment]
@thyeun can you ssh into a machine?

`docker-machine ssh $machineName` and check if the path exist?

**Comment by thyeun** [comment]
@eivantsov , yes, show at the below picture, but i can not access into c/users/

![s12](https://cloud.githubusercontent.com/assets/1939446/17664879/a109b90e-6329-11e6-9a58-3c6512d936e5.JPG)

![s13](https://cloud.githubusercontent.com/assets/1939446/17664943/31aa0efa-632a-11e6-9a9c-52c434c82ead.JPG)

**Comment by thyeun** [comment]
sorry @eivantsov , i typo mistake, i can access it

![s14](https://cloud.githubusercontent.com/assets/1939446/17664994/9c04e40a-632a-11e6-9832-aafc33c332d4.JPG)

**Comment by ghost** [comment]
@thyeun do you see `mount` dir in there?

**Comment by thyeun** [comment]
@eivantsov no, the last one is **conf**

**Comment by TylerJewell** [comment]
@thyeun - if you try to mount into the /conf directory, does that work?  How did you create the /mount directory on your windows host?

**Comment by TylerJewell** [comment]
@thyeun - in reading virtualbox logs - there is also an issue with using capitalized letters and double slashes.  Try `che mount //c/users/thomas/mount 32781' as a test.  If this resolves it - we can do some magic on boot2docker to convert capitalized letters and add a slash.

https://github.com/docker/docker/issues/12590

**Comment by thyeun** [comment]
@TylerJewell the /conf directory is the magic for the customizes che.properties, for github.

ok, i will come back to you once i got the result.

**Comment by thyeun** [comment]
@TylerJewell , `che mount //c/users/thomas/mount 32781' not possible, getting error message **cannot enable tty mode on non tty input**

**Comment by TylerJewell** [comment]
Ok!  That means that the directory is passing for you now.  That is a good thing.  This mount ability is experimental - so we do expect unknown errors.  Try the following.

First - get the IP address for inet from running:

```
docker run --rm --net host alpine sh -c "ip a show eth1"
```

then, run the following

```
docker run --rm -it --cap-add SYS_ADMIN \
                  --device /dev/fuse \
                  -v //c/users/thomas/mount:/mnthost \
                  codenvy/che-mount <ip-addr-from-above> <ws-ssh-port>
```

**Comment by thyeun** [comment]
@TylerJewell Result as show below picture.

![s15](https://cloud.githubusercontent.com/assets/1939446/17669563/b10e5992-6340-11e6-8786-e75172a30fed.JPG)

**Comment by TylerJewell** [comment]
Ooops - bad syntax.  I updated it - replace "docker_exec" with "docker"

**Comment by thyeun** [comment]
@TylerJewell is ok, i figure it out, hahaha, 

Result still same but with different error message, show at below picture

![s16](https://cloud.githubusercontent.com/assets/1939446/17669830/caa71dde-6341-11e6-82c4-6662f9b7161f.JPG)

**Comment by TylerJewell** [comment]
Ok - good - this means that there is flaw with che-mount.  What happens if you replace "-it" with "-i"?

**Comment by thyeun** [comment]
look better, pulling now, but abit slow!!!!

![s17](https://cloud.githubusercontent.com/assets/1939446/17669997/89b6555a-6342-11e6-965f-896ee1dc69ba.JPG)

will come back to you once it finish :)

**Comment by thyeun** [comment]
@TylerJewell , the pulling process not moving at all. about 2 hours.

![s18](https://cloud.githubusercontent.com/assets/1939446/17673475/d9721ece-6353-11e6-9c8a-ffe9077db857.JPG)

**Comment by TylerJewell** [comment]
That is a problem with docker - should be really fast - it's less than 5mb.  kill it and try again.

**Comment by thyeun** [comment]
@TylerJewell Result as show from below picture, an error occur!!!

![s19](https://cloud.githubusercontent.com/assets/1939446/17682143/41cf6a12-637c-11e6-8299-e24f191d314e.JPG)

**Comment by TylerJewell** [comment]
And without the double slash '//'? -- just a single slash '/'

**Comment by thyeun** [comment]
Result **'/ '** as picture below

![s20](https://cloud.githubusercontent.com/assets/1939446/17682325/6198c8b0-637d-11e6-93c5-fb7331fb9eef.JPG)

**Comment by TylerJewell** [comment]
Everything looks good - we'll have to leave this open until an engineer can run some tests with che mount for virtualbox.  That will require a special environment.

**Comment by thyeun** [comment]
ok......thanks

last question before i off, why everytimes i restart my PC and start again Che, my workspace will be gone?

and this **docker run --rm -t -v //var/run/docker.sock:/var/run/docker.sock -e CHE_CONF_FOLDER=//c/Users/thomas/conf eclipse/che start**, can not be working any more, i just manage to make the github auth working one time, after PC restart, it can not be working!!!

**Comment by TylerJewell** [comment]
YOu have the wrong variable :).  You want `CHE_DATA_FOLDER`, not `CHE_CONF_FOLDER`.  See: https://eclipse-che.readme.io/docs/usage-docker#environment-variables

**Comment by thyeun** [comment]
ok... understand for the workspace issue need CHE_DATA_FOLDER.

but for a custom che.properties for github auth,  it should be CHE_CONF_FOLDER, correct me if i'm wrong?

**Comment by TylerJewell** [comment]
Yeah, that is correct.  CHE_CONF_FOLDER is where your che.properties file is. 

If it's not working on PC restart - is it because your VM is not running in virtualbox?  If the VM is running, did virtualbox properly mount the CHE_CONF_FOLDER again?  So if you SSH into the vbox VM, is /c/users/thomas/conf/che.properties there?

**Comment by thyeun** [comment]
Until here /c/users/thomas/conf from SSH into VM, can not find the file che.properties.

**Comment by thyeun** [comment]
@TylerJewell abit different from #2122 , 

**first show you the pictures SSH inside vboxVM, before restart PC, with the command line** 

**docker run --rm -t -v //var/run/docker.sock:/var/run/docker.sock -e CHE_DATA_FOLDER=//c/Users/thomas/mount -e CHE_CONF_FOLDER=//c/Users/thomas/cong eclipse/che start**

**1st** picture (for CHE_DATA_FOLDER), i got a **"blank folder"** inside IDE but SSH inside vboxVM, i can not find it

![s21](https://cloud.githubusercontent.com/assets/1939446/17696504/5e4464c6-63e1-11e6-87fd-6429bb952515.JPG)

**2nd** picture (for CHE_CONF_FOLDER), i create "cong folder" for custom che.properties. but SSH inside vboxVM i can not find the file.

![s22](https://cloud.githubusercontent.com/assets/1939446/17696627/3472c786-63e2-11e6-8a84-7eefc7dc39bd.JPG)

**second show you the pictures SSH inside vboxVM after restart PC, with same command line**

As from the picture below, you notice that the workspace that i created before restart PC, it gone.

![s23](https://cloud.githubusercontent.com/assets/1939446/17696841/a213f480-63e3-11e6-889b-10d4beedf636.JPG)

**Comment by TylerJewell** [comment]
@thyeun - I have been able to reproduce the issue on a local machine.  Labeling it as a bug.

**Comment by TylerJewell** [comment]
@thyeun - in doing a variety of tests on Boot2Docker systems today, we found an issue with the CHE_DATA_FOLDER and CHE_LOCAL_BINARY commands for this particular system. I have submitted this pull request, which has improvements for the che-launcher container and the CLI which will fix the issues permanently.

Things should mostly work on windows as long as you keep your folders in c:\Users AND you do not have spaces in your folders.  I was able to get spaces with folders to work on windows, but it will require a special syntax given some limitations in Docker.

https://github.com/eclipse/che/pull/2119

**Comment by thyeun** [comment]
@TylerJewell i just want to show you the spacing folder outcome also, but since you notice, than i keep the try for next version. hahaha

**Comment by JamesDrummond** [comment]
@TylerJewell Can you provide a status label?

**Comment by TylerJewell** [comment]
@JamesDrummond - I removed the milestone from this issue.  It's not getting worked on currently - will not be until after #2119 is merged and resolved.

**Comment by TylerJewell** [comment]
@thyeun - we did a commit to master today that will fix the CHE_DATA_FOLDER issues for windows.  We still will not support spaces in the folder names, though.

**Comment by TylerJewell** [comment]
@thyeun - closing this issue as a commit was done and it's in master now.

**Comment by JamesDrummond** [comment]
@TylerJewell Will this be included in 4.7 or will we add it to the up coming 4.7.1?

**Comment by TylerJewell** [comment]
This is just a support ticket - so it doesn't get a milestone.
